### PR TITLE
test: add e2e module 

### PR DESCRIPTION
**Pull request by hadothanh**
My proposal:
- Run 2e2 smoke test for each pull request
- using Javascript for more familiar with dev team
- Add more testing check point for FE deployment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -149,7 +152,11 @@
     "@types/wcag-contrast": "^3.0.0",
     "@typescript-eslint/eslint-plugin": "^4.33.0",
     "@typescript-eslint/parser": "^4.33.0",
+    "browserslist": "^4.21.4",
+    "caniuse-lite": "^1.0.30001439",
```
Comment (viet-nv) [inline package.json:156]: what is the purpose of these packages?

File: .env [hunk 2]
```diff
@@ -35,4 +35,8 @@ REACT_APP_GOOGLE_RECAPTCHA_KEY=6Lc7yg0iAAAAALhS4C4Ez5hqMBub8hKdFvvr9sfc
 REACT_APP_POOL_FARM_BASE_URL=https://pool-farm.kyberswap.com
 
 REACT_APP_LIMIT_ORDER_API_WRITE=https://limit-order.dev.kyberengineering.io/write/api
-REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.dev.kyberengineering.io/read-ks/api
\ No newline at end of file
+REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.dev.kyberengineering.io/read-ks/api
+
+SECRET_WORDS="test, test, test, test, test, test, test, test, test, test, test, junk"
```
Comment (namgold) [inline .env:40]: add to .env.stg too

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -106,12 +106,13 @@ jobs:
 
       - name: Run Lint
         run: yarn lint --quiet
-
+
```
Comment (namgold) [inline .github/workflows/pr.yaml:109]:
```suggestion

```

File: package.json [hunk 4]
```diff
@@ -158,9 +165,13 @@
...
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-unused-imports": "^2.0.0",
+    "mochawesome": "^7.1.3",
     "prettier": "^2.7.1",
     "react-scripts": "^4.0.3",
     "source-map-explorer": "^2.5.3",
     "typescript": "^4.8.4"
+  },
+  "resolutions": {
+    "ethers": "^5.0.0"
```
Comment (namgold) [inline package.json:175]: seems dangerous

File: synpress/e2e/pages/swap/swap.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  getTokenSearchByName(name) {
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(700)
+    return cy.get(`div[title='${name}']`).first()
+  }
+
+  getRoutePercentLabel() {
+    return cy.xpath("//div[contains(@class,'StyledPercent')]")
```
Comment (namgold) [inline synpress/e2e/pages/swap/swap.js:31]: why dont `cy.get('div.StyledPercent')`?

File: synpress/e2e/specs/swap.spec.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    swapPage.waitUntilLoggedIn()
+  })
+
+  context('Connect metamask wallet', () => {
+    it('should connect with success', () => {
+      swapPage.getLoggedInWalletAddress().then(stakingWalletAddress => {
+        const formattedMetamaskWalletAddress =
+          metamaskWalletAddress.slice(0, 6) + '...' + metamaskWalletAddress.slice(-4)
+        expect(stakingWalletAddress.toLowerCase()).to.equal(formattedMetamaskWalletAddress.toLowerCase())
+        debugger
```
Comment (namgold) [inline synpress/e2e/specs/swap.spec.js:34]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (hadothanh) [reply package.json]: I removed them

File: package.json (hunk 4 above)
Comment (hadothanh) [reply package.json:173]: @namgold : the libraries for test module

Comment (namgold) [reply package.json:173]: This might causing break since this impacts the the codebase. We need to test this carefully.

File: .env [hunk 7]
```diff
@@ -37,5 +37,10 @@ REACT_APP_POOL_FARM_BASE_URL=https://pool-farm.kyberswap.com
 REACT_APP_LIMIT_ORDER_API_WRITE=https://limit-order.dev.kyberengineering.io/write/api
 REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.dev.kyberengineering.io/read-ks/api
 
+<<<<<<< HEAD
```
Comment (nguyenhoaidanh) [inline .env:40]: conflict

File: .env.stg [hunk 8]
```diff
@@ -35,6 +35,11 @@ REACT_APP_POOL_FARM_BASE_URL=https://pool-farm.kyberswap.com
 
 REACT_APP_LIMIT_ORDER_API_WRITE=https://limit-order.stg.kyberengineering.io/write/api
 REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.stg.kyberengineering.io/read-ks/api
+<<<<<<< HEAD
```
Comment (nguyenhoaidanh) [inline .env.stg:38]: conflict

File: synpress/e2e/pages/swap/header.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  }
+  getAcceptCheckbox() {
+    return cy.get('[type="checkbox"]')
+  }
+  getMetaMaskWalletOption() {
+    return cy.get('#connect-METAMASK')
+  }
+
+  getWalletBalance() {
+    cy.get('#selectNetwork > div > div > div')
```
Comment (nguyenhoaidanh) [inline synpress/e2e/pages/swap/header.js:25]: need to put id into balance label

**Comment by nguyenhoaidanh** [comment]
review done.

**Pull request closed without merging**